core/perf_test/test_atomic_minmax_simple.cpp

```diff
@@ -0,0 +1,221 @@
+// export OMP_PROC_BIND=spread ; export OMP_PLACES=threads
+// c++  -O2 -g -DNDEBUG  -fopenmp
+// ../core/perf_test/test_atomic_minmax_simple.cpp -I../core/src/ -I. -o
+// test_atomic_minmax_simple.x  containers/src/libkokkoscontainers.a
+// core/src/libkokkoscore.a -ldl && OMP_NUM_THREADS=1
+// ./test_atomic_minmax_simple.x 10000000
+
+#include <cstdio>
+#include <cstdlib>
+
+#include <iostream>
+#include <typeinfo>
+
+#include <Kokkos_Core.hpp>
+#include <impl/Kokkos_Timer.hpp>
+
+using exec_space = Kokkos::DefaultExecutionSpace;
+
+template <typename T>
+void sanity(void) {
+  T a = 0;
+  T b = 1;
+  T c;
+
+  std::cout << "atomic_fetch_min:\n";
+  c = Kokkos::atomic_fetch_min(&a, b);
+  std::cout << "a=" << a << "\n";
+  std::cout << "b=" << b << "\n";
+  std::cout << "c=" << c << "\n";
+
+  std::cout << "atomic_fetch_max:\n";
+  c = Kokkos::atomic_fetch_max(&a, b);
+  std::cout << "a=" << a << "\n";
+  std::cout << "b=" << b << "\n";
+  std::cout << "c=" << c << "\n";
+
+  std::cout << "atomic_fetch_max:\n";
+  c = Kokkos::atomic_fetch_max(&a, b);
+  std::cout << "a=" << a << "\n";
+  std::cout << "b=" << b << "\n";
+  std::cout << "c=" << c << "\n";
+}
+
+template <typename T>
+void test(const int length) {
+  Kokkos::Impl::Timer timer;
+
+  typedef Kokkos::View<T*, exec_space> vector;
+
+  vector inp("input", length);
+
+  // input is max values - all min atomics will replace
+  {
+    T max = std::numeric_limits<T>::max();
+
+    Kokkos::parallel_for(
+        length, KOKKOS_LAMBDA(const int i) { inp(i) = max; });
+    Kokkos::fence();
```

Comment:
You can achieve this via `Kokkos::deep_copy(inp, max);`

Reply:
That's useful.  I just prefer general-purpose tools to specific ones (https://slate.com/human-interest/2015/12/alton-brown-is-right-to-scorn-single-use-kitchen-gadgets-and-the-people-who-buy-them.html).